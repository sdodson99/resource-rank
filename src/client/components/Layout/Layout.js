import React from 'react';
import PropTypes from 'prop-types';
import styles from './Layout.module.css';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import useReadOnlyModeEnabledQuery from '@/hooks/queries/use-read-only-mode-enabled-query';

const Layout = ({ children }) => {
  const readOnlyModeEnabled = useReadOnlyModeEnabledQuery();

  return (
    <div className={styles.Layout} data-testid="Layout">
      <div className={styles.Page}>
        {readOnlyModeEnabled && (
          <div className={styles.Alert}>Application is in read-only mode.</div>
        )}
        <Header />
        <div>{children}</div>
      </div>
      <Footer />
    </div>
  );
};

Layout.propTypes = {
  children: PropTypes.node,
};

Layout.defaultProps = {};

export default Layout;
