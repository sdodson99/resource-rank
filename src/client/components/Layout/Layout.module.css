.Layout {
}

.Page {
  min-height: 100vh;
}

.Alert {
  color: #664d03;

  @apply bg-yellow-100 py-4 text-center;
}
