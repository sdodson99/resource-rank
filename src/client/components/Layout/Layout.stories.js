import React from 'react';
import Layout from './Layout';

export default {
  component: Layout,
  title: 'Components/Layout',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <Layout {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: <div className="my-5 content-container">Hello world!</div>,
};
