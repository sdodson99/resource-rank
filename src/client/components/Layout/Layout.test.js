import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Layout from './Layout';
import useReadOnlyModeStatus from '../../hooks/use-read-only-mode-status';
import renderer from 'react-test-renderer';

jest.mock('../../hooks/use-read-only-mode-status');
jest.mock('../../hooks/use-firebase-app');

describe('<Layout />', () => {
  beforeEach(() => {
    useReadOnlyModeStatus.mockReturnValue(true);
  });

  afterEach(() => {
    useReadOnlyModeStatus.mockReset();
  });

  it('should mount', () => {
    render(<Layout />);

    const layout = screen.getByTestId('Layout');

    expect(layout).toBeInTheDocument();
  });

  describe('with read only mode enabled', () => {
    beforeEach(() => {
      useReadOnlyModeStatus.mockReturnValue(true);
    });

    afterEach(() => {
      useReadOnlyModeStatus.mockReset();
    });

    it('should display read only mode', () => {
      render(<Layout />);

      const readOnlyModeElement = screen.queryByText(
        'Application is in read-only mode.'
      );

      expect(readOnlyModeElement).toBeTruthy();
    });

    it('should render correctly', () => {
      const tree = renderer.create(<Layout />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('with read only mode disabled', () => {
    beforeEach(() => {
      useReadOnlyModeStatus.mockReturnValue(false);
    });

    afterEach(() => {
      useReadOnlyModeStatus.mockReset();
    });

    it('should not display read only mode', () => {
      render(<Layout />);

      const readOnlyModeElement = screen.queryByText(
        'Application is in read-only mode.'
      );

      expect(readOnlyModeElement).toBeNull();
    });

    it('should render correctly', () => {
      const tree = renderer.create(<Layout />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
