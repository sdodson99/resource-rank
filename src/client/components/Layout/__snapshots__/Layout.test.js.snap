// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> with read only mode disabled should render correctly 1`] = `
<div
  className="Layout"
  data-testid="Layout"
>
  <div
    className="page"
  >
    <header
      className="Header"
      data-testid="Header"
    >
      <div>
        <div>
          <div>
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "block",
                      "maxWidth": "100%",
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzUiIGhlaWdodD0iNzUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style={
                      Object {
                        "border": "none",
                        "display": "block",
                        "margin": 0,
                        "maxWidth": "100%",
                        "padding": 0,
                      }
                    }
                  />
                </div>
                <img
                  alt="Resource Rank Logo"
                  decoding="async"
                  src="/_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75"
                  srcSet="/_next/image?url=%2Fimg%2Flogo.svg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div>
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </div>
          <div>
            <a
              href="/topics"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Topics
            </a>
          </div>
          <div>
            <button
              onClick={[Function]}
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </header>
    <div />
  </div>
  <footer
    className="Footer"
    data-testid="Footer"
  >
    <div>
      <div>
        <div>
          <h1>
            Our Mission
          </h1>
          <p>
            is to 
            <strong>
              simplify
            </strong>
             your learning journey. Learning can be 
            <strong>
              difficult
            </strong>
            , especially when starting. There are 
            <strong>
              millions
            </strong>
             of educational courses, videos, and books 
            <strong>
              scattered
            </strong>
             across the Internet.
          </p>
          <p>
            Resource Rankâ€™s goal is to 
            <strong>
              consolidate
            </strong>
             all of these resources so that you can 
            <strong>
              quickly
            </strong>
             find the
             
            <strong>
              best
            </strong>
             resources for what you are learning.
          </p>
        </div>
        <div>
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "inline-block",
                "margin": 0,
                "maxWidth": "100%",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "block",
                  "maxWidth": "100%",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzUiIGhlaWdodD0iNzUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "padding": 0,
                  }
                }
              />
            </div>
            <noscript>
              <img
                alt="Resource Rank Logo"
                decoding="async"
                src="/img/logo.svg"
                srcSet="/_next/image?url=%2Fimg%2Flogo.svg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
            <img
              alt="Resource Rank Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`<Layout /> with read only mode enabled should render correctly 1`] = `
<div
  className="Layout"
  data-testid="Layout"
>
  <div
    className="page"
  >
    <div>
      Application is in read-only mode.
    </div>
    <header
      className="Header"
      data-testid="Header"
    >
      <div>
        <div>
          <div>
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "display": "block",
                      "maxWidth": "100%",
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzUiIGhlaWdodD0iNzUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style={
                      Object {
                        "border": "none",
                        "display": "block",
                        "margin": 0,
                        "maxWidth": "100%",
                        "padding": 0,
                      }
                    }
                  />
                </div>
                <img
                  alt="Resource Rank Logo"
                  decoding="async"
                  src="/_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75"
                  srcSet="/_next/image?url=%2Fimg%2Flogo.svg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div>
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </div>
          <div>
            <a
              href="/topics"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Topics
            </a>
          </div>
          <div>
            <button
              onClick={[Function]}
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </header>
    <div />
  </div>
  <footer
    className="Footer"
    data-testid="Footer"
  >
    <div>
      <div>
        <div>
          <h1>
            Our Mission
          </h1>
          <p>
            is to 
            <strong>
              simplify
            </strong>
             your learning journey. Learning can be 
            <strong>
              difficult
            </strong>
            , especially when starting. There are 
            <strong>
              millions
            </strong>
             of educational courses, videos, and books 
            <strong>
              scattered
            </strong>
             across the Internet.
          </p>
          <p>
            Resource Rankâ€™s goal is to 
            <strong>
              consolidate
            </strong>
             all of these resources so that you can 
            <strong>
              quickly
            </strong>
             find the
             
            <strong>
              best
            </strong>
             resources for what you are learning.
          </p>
        </div>
        <div>
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "inline-block",
                "margin": 0,
                "maxWidth": "100%",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "block",
                  "maxWidth": "100%",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzUiIGhlaWdodD0iNzUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "margin": 0,
                    "maxWidth": "100%",
                    "padding": 0,
                  }
                }
              />
            </div>
            <noscript>
              <img
                alt="Resource Rank Logo"
                decoding="async"
                src="/img/logo.svg"
                srcSet="/_next/image?url=%2Fimg%2Flogo.svg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
            <img
              alt="Resource Rank Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
